pkg/sqlcache/informer/listoption_indexer.go

```diff
@@ -1044,3 +1075,38 @@ func toUnstructuredList(items []any) *unstructured.UnstructuredList {
 	}
 	return result
 }
+
+func matchWatch(filterName string, filterNamespace string, filterSelector string, oldObj any, obj any) bool {
+	matchOld := false
+	if oldObj != nil {
+		matchOld = matchFilter(filterName, filterNamespace, filterSelector, oldObj)
+	}
+	return matchOld || matchFilter(filterName, filterNamespace, filterSelector, obj)
+}
+
+func matchFilter(filterName string, filterNamespace string, filterSelector string, obj any) bool {
+	if obj == nil {
+		return false
+	}
+	metadata, err := meta.Accessor(obj)
+	if err != nil {
+		return false
+	}
+	if filterName != "" && filterName != metadata.GetName() {
+		return false
+	}
+	if filterNamespace != "" && filterNamespace != metadata.GetNamespace() {
+		return false
+	}
+	if filterSelector != "" {
+		selector, err := labels.Parse(filterSelector)
+		if err != nil {
+			fmt.Println("error parsing selector", err)
```

Comment:
if fmt.Println intentional here?

Reply:
Indeed it is not. Thanks for catching this! Removed. And moved the validation of the selector higher up the stack.